.bazelrc

```diff
@@ -128,6 +131,10 @@ build:linux --action_env=BAZEL_LINKLIBS='-l%:libc++.a -lm -static-libgcc'
 # https://github.com/bazelbuild/bazel/issues/12439#issuecomment-914449079
 build:linux --force_pic
 
+# On Linux, garbage collection sections and optimize binary size. These do not apply to the macOS
+# toolchain.
+build:linux --linkopt="-Wl,--gc-sections" --linkopt="-Wl,-O2"
```

Comment:
Do we want these to be on for all Linux builds?  gc-sections shouldn't really be slower, but will only be somewhat effective if `-ffunction-sections -fdata-sections` is not enabled during compilation. All three of these are enabled by default in bazel's opt builds. `-Wl,-O2` will slow down linking somewhat, perhaps it's sufficient to only turn it on for opt builds (and debug builds if needed to stay within CI limits). See my upcoming [llvm-15 PR](https://github.com/cloudflare/workerd/compare/main...felix/llvm-15#diff-ae32976a96fc56a73a2a3c76ad54e4c2f162204f4342871a5bf309d6367dfdc2R26) on how to only turn on linker flags for some Linux builds – it adds `-Wl,--icf=safe` for opt builds which is supported starting with that version.

Reply:
Thanks felix. 

I found --icf=safe to be super chatty in the build log and a 1% space saving (it appeared to be working as intended). From experimenting with different options, the simple-template was the biggest space saver, but I don't think I had the sections flags set inconjunction with `--gc-sections`. I'll give that a try.